databricks/sdk/oidc_token_supplier.py

```diff
@@ -26,3 +29,60 @@ def get_oidc_token(self, audience: str) -> Optional[str]:
             return None
 
         return response_json["value"]
+
+
+class AzureDevOpsOIDCTokenSupplier:
+    """
+    Supplies OIDC tokens from Azure DevOps pipelines.
+
+    Constructs the OIDC token request URL using official Azure DevOps predefined variables.
+    See: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables
+    """
+
+    def get_oidc_token(self, audience: str) -> Optional[str]:
+        # Note: Azure DevOps OIDC tokens have a fixed audience of "api://AzureADTokenExchange"
+        # The audience parameter is ignored but kept for interface compatibility with other OIDC suppliers
+
+        access_token = os.environ.get("SYSTEM_ACCESSTOKEN")
+        collection_uri = os.environ.get("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI")
+        project_id = os.environ.get("SYSTEM_TEAMPROJECTID")
+        plan_id = os.environ.get("SYSTEM_PLANID")
+        job_id = os.environ.get("SYSTEM_JOBID")
+        hub_name = os.environ.get("SYSTEM_HOSTTYPE")
+
+        # Check for required variables
+        if not all([access_token, collection_uri, project_id, plan_id, job_id, hub_name]):
+            # not in Azure DevOps pipeline
+            logger.debug("Azure DevOps OIDC: not in Azure DevOps pipeline environment")
+            return None
+
+        try:
+            # Construct the OIDC token request URL
```

Comment:
```suggestion
            # Construct the OIDC token request URL.
```
Comment should be proper sentence ending with a period.
+ ditto all